Modules/atexitmodule.c

```diff
@@ -12,6 +12,16 @@
 #include "pycore_interp.h"        // PyInterpreterState.atexit
 #include "pycore_pystate.h"       // _PyInterpreterState_GET
 
+#ifdef Py_GIL_DISABLED
+/* Note: anything declared as static in this file assumes the lock is held
```

Comment:
atexit_run_exitfuncs() acquires the lock, whereas atexit_callfuncs() assumes that it's called with the lock held.

It may more explicit to add `_locked` suffix to functions which should be called with the lock held. What do you think?

Reply:
I guess that works. It might be misleading for the default build because there's no locking happening there, but I don't think it will be an issue. Will change the names.